### ISSUE TITLE
Curly doesn't work with empty loops

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#3077]
With the following code:

```js
/* eslint curly: 2*/
for (var i = 0; i <= 5; console.log(i++));
```

You get the curly error:

```sh
2:0 Expected { after 'for' condition. (curly)
```

It should not error for a body-less loops.

This is reproducible in online demo (so, version 0.24.1)

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by btmills** [comment]
Isn't this case handled by the "multi" and "multi-line" modes? Using either of those in the demo removes the warning for me, and based on the docs, this case is exactly what curly's "all" mode (default) is designed to catch.

**Comment by ilyavolodin** [comment]
I don't think so. Default configuration for curly should warn if you didn't use curly braces around a single statement body of loops or conditions, but in this case, there is no body at all. If I would like to use default configuration (as in, always require curly braces around any block-like based construct), I should be able to do so, when they body doesn't exist.

**Comment by btmills** [comment]
Ah, so you're saying there should be a special case for when the body is just an `EmptyStatement`, so that the following snippet would be valid?

```js
/* eslint no-extra-semi: 2 */
/* eslint curly: 2 */

for (var i = 0; i < 10; i++);
```

(Right now in the demo it has two, one from `no-extra-semi`, which will be removed by #3076, and the other from `curly`.)

**Comment by michaelficarra** [comment]
> but in this case, there is no body at all

That's really not true at all. The body is a statement. Sometimes it's a block statement which contains other statements. This one is basically the no-op statement.

**Comment by ilyavolodin** [comment]
@michaelficarra Well sure. But the bottom line, requesting curly braces after this type of loop is incorrect, as there's nothing that should be wrapped into those braces.

@btmills Correct. I think curly shouldn't complain about loops with `EmptyStatment` as a body under any of the available options.

**Comment by lo1tuma** [comment]
I think the current behavior of this rule is correct. You should write `for (var i = 0; i <= 5; console.log(i++)) {}` instead.

**Comment by btmills** [comment]
I'm with @lo1tuma on this - we need a way to prohibit accidental empty bodies, so the current behavior works.

**Comment by nzakas** [comment]
Agree with @lo1tuma - this rule is supposed to force you to use blocks, so this warning is expected.

**Comment by ilyavolodin** [comment]
Ok, sounds good. Closing.
